### PR TITLE
[ODOO-414][ADD] Added new view description to control panel area

### PR DESCRIPTION
**Pull request by Jessica-BlueStingray**
### **[Add report description](https://metricwise.atlassian.net/browse/ODOO-414)**
This update adds a description area and `view_description` field to the Control Panel between the name and the Search Area.

To test:
Upgrade the modules
Turn on debug mode
Click on the debug button > Edit Action > Add a text to `view_description` > Save > Refresh page.

### CODE REVIEW

**Comment by Jessica-BlueStingray** [comment]
@amh-mw Travis reminded me that I should had made the `view_description` feature responsive to different screen sizes. I just added a commit that hides it when the screen is smaller. See video:  


https://user-images.githubusercontent.com/87671101/188955443-cb577089-6ba8-476b-94cd-90e6245c1b79.mp4

**Comment by Jessica-BlueStingray** [comment]
@amh-mw PR is down to 1 commit now

**Comment by amh-mw** [comment]
This doesn't seem to want to patch with either -p1 or -p2. This is an unfortunate side effect of how Odoo changes the file hierarchy on disk as part of the Python packaging process. The "easy" fix would be to break this into two pull requests and patch one each way, but... 🤮 Let me think about it.

**Review by amh-mw — DISMISSED** [review]
So after carefully reviewing the patch -p1 output, I think the best course of action is to take the server side changes to ir.actions and move them into a separate odoo-metricwise pull requests -- see my untested ODOO-414 branch for a head start.

File: odoo/addons/base/views/ir_actions_views.xml [hunk 1]
```diff
@@ -420,7 +421,7 @@ env['res.partner'].create({'name': partner_name})
                   <header>
                         <button name="action_launch"
                             states="open" string="Launch"
-                            type="object" icon="fa-cogs" class="oe_highlight" 
+                            type="object" icon="fa-cogs" class="oe_highlight"
```
Comment (amh-mw) [inline odoo/addons/base/views/ir_actions_views.xml:424]: Nit: No whitespace cleanup since we're never going to merge this upstream.

**Comment by amh-mw** [comment]
Just as a heads up, I am working up a new patching modality that may solve this without any changes. Stay tuned.

**Comment by Jessica-BlueStingray** [comment]
> Just as a heads up, I am working up a new patching modality that may solve this without any changes. Stay tuned.

Sorry I was really busy. 
I'm currently waiting for the news on the new patching then. If that doesn't solve it I'll do the other changes required.

**Comment by amh-mw** [comment]
Ok, so the latest develop I just pushed adds a new `patch.sh` script that uses the GitHub CLI tool `gh` to figure out what patches to pull, then `sed` to modify their path hints so that -p1 works for all of them.

To test your patch, you can basically extract 

```
DIR="docker/odoo/patches"
NUM="20"
URL="https://patch-diff.githubusercontent.com/raw/MetricWise/odoo/pull"

curl --output-dir $DIR --remote-name --silent --url "$URL/$NUM.diff"
sed -E -I '' -e 's%^(--- a|\+\+\+ b)/odoo%\1%' "$DIR/$NUM.diff"
```

And then `./run.sh -b` or `./test.sh -b` and if it looks good, `git add -u docker/odoo/patches/20.diff`

**Comment by Jessica-BlueStingray** [comment]
@amh-mw The file was added correctly but I can't get past the `sed` command, I believe that's due to platform dependency, I'm using **Linux**. So I couldn't finish testing yet. Is there a `sed` command _linux version_ I can use?

**Comment by traviswaelbro** [comment]
I believe Mac uses a slightly different version of `sed` (and similar commands) compared to Linux. There's a [`gnu-sed`](https://formulae.brew.sh/formula/gnu-sed) that can be installed with Brew that should run the same `sed` as Linux. It's installed as `gsed` by default, but can be aliased to `sed` if desired.

**Comment by amh-mw** [comment]
I can't say that I have ever used any unix flavor that didn't have [sed](https://linux.die.net/man/1/sed). It's been around [since 1973](https://en.wikipedia.org/wiki/Sed). 😁

Centos 6:
```
$ yum whatprovides `which sed`
sed-4.2.2-5.el7.x86_64 : A GNU stream text editor
```

Debian 11:
```
$ dpkg -l sed
sed            4.7-1        amd64        GNU stream editor for filtering/transforming text
```

**Comment by amh-mw** [comment]
Ah... so the issue is that your flavor of sed doesn't have -E for ERE? I'll just have to rewrite the pattern as BRE.

**Comment by Jessica-BlueStingray** [comment]
I'm not sure what the issue is, but the `sed` is not locating the file, although the file has been added correctly. 
_I think_ it is due to how the command is written, it might need to be different for linux. Sorry for my poor explanation earlier.
Screenshot from 2022-09-16 14-59-48

**Comment by amh-mw** [comment]
I've force pushed a BRE-compatible script to develop. Could you type `which sed`

**Comment by Jessica-BlueStingray** [comment]
> I've force pushed a BRE-compatible script to develop. Could you type `which sed`

Yes, I get `/usr/bin/sed`

**Comment by amh-mw** [comment]
Try the new patterns?
```
sed -i '' -e 's%^\(--- a\)/odoo%\1%' "$DIR/$NUM.diff"
sed -i '' -e 's%^\(+++ b\)/odoo%\1%' "$DIR/$NUM.diff"
```

**Comment by Jessica-BlueStingray** [comment]
Still nothing

**Comment by amh-mw** [comment]
What flavor of unix are you running? Maybe I can stand up a container to try and reproduce?

**Comment by Jessica-BlueStingray** [comment]
Linux 5.15.0-46-generic
Ubuntu 20.04.4 LTS

**Comment by amh-mw** [comment]
```
docker run ubuntu:focal sed --help

  -E, -r, --regexp-extended
                 use extended regular expressions in the script
                 (for portability use POSIX -E).
```

So you do have access to ERE. Maybe that's not the problem. Is your source code volume mounted read-only?

**Comment by Jessica-BlueStingray** [comment]
> Is your source code volume mounted read-only?

No, it's not

**Comment by Jessica-BlueStingray** [comment]
@amh-mw is there anything else I can to do to help here?

**Comment by Jessica-BlueStingray** [comment]
@amh-mw  we managed it. It ran successfully.
The `sed` used in Ubuntu is `gsed (GNU)` for MAC, and command line used was slightly different than the original:

original: `sed -E -I '' -e 's%^(--- a|\+\+\+ b)/odoo%\1%' "$DIR/$NUM.diff"`
used:    `sed -E -i'' -e 's%^(--- a|\+\+\+ b)/odoo%\1%' "$DIR/$NUM.diff"`

We removed the silent flag from the `curl` command line and found out that it was silently failing too due to `https` not being supported, then we changed to `http` and it was ok after that.

Both test and run were successful. Is there anything else we can do?

**Comment by amh-mw** [comment]
You should create a pull request against odoo-metricwise, now that your patch has been copied into the docker/odoo/patches folder.

**Comment by Jessica-BlueStingray** [comment]
> You should create a pull request against odoo-metricwise, now that your patch has been copied into the docker/odoo/patches folder.

I don't think I have permissions..

Screenshot from 2022-09-22 17-00-55

**Comment by traviswaelbro** [comment]
@Jessica-BlueStingray I think Adam is referring to a new PR in the other repo, similar to Dylan's on MetricWise/odoo-metricwise#8

**Comment by Jessica-BlueStingray** [comment]
Like this? : [Metricwise#11](https://github.com/MetricWise/odoo-metricwise/pull/11)

**Comment by amh-mw** [comment]
Moved into #27.

**Pull request closed without merging**